From c9539c76028a8bffa0bfc73dbcdd7e18f3f858f4 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Fri, 1 May 2020 08:01:18 -0500
Subject: [PATCH 25/25] arm64: dts: beacon-imx8mm: Fix USB Host Reset

The USB controller is plumbed through a no-op USB tranceiver which
really only has a few purposes, but one of them is to be able to
reset the device.

Until now, the USB has been held out of reset using a fixed-regulator,
but this is incorrect.

This patch moves the gpio functions out of the USB's fixed regulator
into the usbphynop2 node which allows the USB system to reset the hub
as needed.  This also appears to fix an issue where the USB fails
to recover from overcurrent.

Fixes: PHOENIX8-78

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
index d84c4d758f75..a3279847d711 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
@@ -121,8 +121,6 @@
 		regulator-name = "usb_h1_vbus";
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
-		gpio = <&pca6416_1 7 GPIO_ACTIVE_LOW>; /* nRST */
-		enable-active-low;
 	};
 
 	reg_audio: regulator-audio {
@@ -524,6 +522,10 @@
 	status = "okay";
 };
 
+&usbphynop2 {
+	reset-gpios = <&pca6416_1 7 GPIO_ACTIVE_HIGH>;
+};
+
 /*rfee  from uboot*/
 &usdhc2 {
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-- 
2.17.1

